rebasehelper/logger.py

```diff
@@ -157,38 +160,58 @@ def add_file_handler(logger: logging.Logger, path: str, formatter: Optional[logg
-        results_dir: Path to rebase-helper-results directory.
+        Returns:
+            List of created file handler instances.
 
-    """
-    logs_dir = os.path.join(results_dir, constants.LOGS_DIR)
-    logger = logging.getLogger('rebasehelper')
+        """
+        logs_dir = os.path.join(results_dir, constants.LOGS_DIR)
+        logger = logging.getLogger('rebasehelper')
 
-    log_formatter = logging.Formatter('%(message)s')
-    debug_log_formatter = logging.Formatter('%(asctime)s %(filename)s:%(lineno)s %(funcName)s: %(message)s')
+        log_formatter = logging.Formatter('%(message)s')
+        debug_log_formatter = logging.Formatter('%(asctime)s %(filename)s:%(lineno)s %(funcName)s: %(message)s')
 
-    debug_log = os.path.join(logs_dir, constants.DEBUG_LOG)
-    LoggerHelper.add_file_handler(logger, debug_log, debug_log_formatter, logging.DEBUG)
-    verbose_log = os.path.join(logs_dir, constants.VERBOSE_LOG)
-    LoggerHelper.add_file_handler(logger, verbose_log, log_formatter, CustomLogger.VERBOSE)
-    info_log = os.path.join(logs_dir, constants.INFO_LOG)
-    LoggerHelper.add_file_handler(logger, info_log, log_formatter, logging.INFO)
+        debug_log = os.path.join(logs_dir, constants.DEBUG_LOG)
+        debug = cls.add_file_handler(logger, debug_log, debug_log_formatter, logging.DEBUG)
+        verbose_log = os.path.join(logs_dir, constants.VERBOSE_LOG)
+        verbose = cls.add_file_handler(logger, verbose_log, log_formatter, CustomLogger.VERBOSE)
+        info_log = os.path.join(logs_dir, constants.INFO_LOG)
+        info = cls.add_file_handler(logger, info_log, log_formatter, logging.INFO)
 
+        return [h for h in (debug, verbose, info) if h]
 
-def create_stream_handlers() -> Tuple[ColorizingStreamHandler, ColorizingStreamHandler]:
-    logger = logging.getLogger('rebasehelper')
-    formatter = logging.Formatter('%(levelname)s: %(message)s')
-    main = LoggerHelper.add_stream_handler(logger, logging.INFO, formatter)
+    @classmethod
+    def create_stream_handlers(cls) -> Tuple[ColorizingStreamHandler, ColorizingStreamHandler]:
+        logger = logging.getLogger('rebasehelper')
+        formatter = logging.Formatter('%(levelname)s: %(message)s')
+        main = cls.add_stream_handler(logger, logging.INFO, formatter)
 
-    logger_summary = logging.getLogger('rebasehelper.summary')
-    logger_summary.propagate = False
-    summary = LoggerHelper.add_stream_handler(logger_summary)
+        logger_summary = logging.getLogger('rebasehelper.summary')
+        logger_summary.propagate = False
+        summary = cls.add_stream_handler(logger_summary)
 
-    logger_report = logging.getLogger('rebasehelper.report')
-    logger_report.propagate = False
-    return main, summary
+        logger_report = logging.getLogger('rebasehelper.report')
+        logger_report.propagate = False
+        return main, summary
+
+    @classmethod
+    def remove_file_handlers(cls, handlers: List[logging.FileHandler]) -> None:
```

Comment:
Shouldn't this rather be called `remove_handlers`? Since it removes every handler that is passed and even stream handlers could be removed.

Reply:
True, but it does that only for `rebasehelper` logger. You can't remove default summary stream handler this way.